src/sentry/static/sentry/app/utils/integrationUtil.tsx

```diff
@@ -45,17 +67,25 @@ type IntegrationsEventParams = (MultipleIntegrationsEvent | SingleIntegrationEve
  */
 export const trackIntegrationEvent = (
   analtyicsParams: IntegrationsEventParams,
-  org: Organization,
+  org?: Organization, //we should pass in org whenever we can but not every place guarantees this
   options?: {startSession: boolean}
 ) => {
   const {startSession} = options || {};
   const sessionId = startSession ? startAnalyticsSession() : getAnalyticsSessionId();
   const fullParams = {
     analytics_session_id: sessionId,
-    organization_id: org.id,
-    role: org.role,
+    organization_id: org && org.id,
```

Comment:
Can we use optional chaining on line 77 & 78?

Reply:
@NisanthanNanthakumar When this PR started, the answer was no. But now we can :)

Reply:
@NisanthanNanthakumar Just kidding...the optional chaining PR has not been merged yet. If it gets merged before this PR gets approved I will add the change.

Reply:
sounds good